lib/fs.js

```diff
@@ -546,9 +546,6 @@ function read(fd, buffer, offset, length, position, callback) {
 
   validateOffsetLengthRead(offset, length, buffer.byteLength);
 
-  if (position == null)
-    position = -1;
-
```

Comment:
We should actually keep this with `if (position === null)` given that the default assignment will not work when `position` is explicitly passed as `null`.

Reply:
@jasnell Note this was intentional, see https://github.com/nodejs/node/pull/37101#discussion_r565347781 above.

Reply:
@jasnell added the check back in along with a deprecation warning.